assets/src/components/with-cropped-featured-image.js

```diff
@@ -1,19 +1,170 @@
+
+		const state = this.controller.state();
+		const selection = state.get( 'selection' );
+
+		const attachment = selection.models[ 0 ];
+		const minWidth = state.collection.get( 'library' ).get( 'suggestedWidth' );
+		const minHeight = state.collection.get( 'library' ).get( 'suggestedHeight' );
+
+		if ( ! attachment || ( attachment.get( 'width' ) >= minWidth && attachment.get( 'height' ) >= minHeight ) ) {
+			this.secondary.unset( 'select-error' );
+		} else {
+			this.secondary.set(
+				'select-error',
+				new FeaturedImageSelectionError( {
+					minWidth,
+					minHeight,
+					width: attachment.get( 'width' ),
+					height: attachment.get( 'height' ),
+				} )
+			);
+
+			if ( this._views ) {
+				for ( const button of Object.values( this._views ) ) {
+					if ( button.model ) {
+						button.model.set( 'disabled', true );
+					}
+				}
+			}
+		}
+	},
+} );
 
 /**
- * Gets a wrapped version of MediaUpload.
+ * FeaturedImageSelectMediaFrame
+ *
+ * Select a featured image from the media library.
+ *
+ * @class
+ * @augments wp.media.view.MediaFrame.Select
+ * @augments wp.media.view.MediaFrame
+ * @augments wp.media.view.Frame
+ * @augments wp.media.View
+ * @augments wp.Backbone.View
+ * @augments Backbone.View
+ * @mixes wp.media.controller.StateMachine
+ * @inheritDoc
+ */
+const FeaturedImageSelectMediaFrame = wp.media.view.MediaFrame.Select.extend( {
+
+	/**
+	 * Create select toolbar.
+	 *
+	 * The only reason for this method is to override the select toolbar view class.
+	 *
+	 * @param {Object} toolbar
+	 * @param {Object} [options={}]
+	 * @this wp.media.controller.Region
+	 */
+	createSelectToolbar( toolbar, options ) {
+		options = options || this.options.button || {};
+		options.controller = this;
+
+		toolbar.view = new FeaturedImageToolbarSelect( options );
+	},
+} );
+
+/**
+ * A state for cropping a featured image.
+ *
+ * @constructs FeaturedImageCropper
+ * @inheritDoc
+ */
+const FeaturedImageCropper = wp.media.controller.Cropper.extend( {
+	/**
+	 * Creates an object with the image attachment and crop properties.
+	 *
+	 * @param {wp.media.model.Attachment} attachment The attachment to crop.
+	 * @return {jQuery.promise} A jQuery promise that represents the crop image request.
+	 */
+	doCrop( attachment ) {
+		const cropDetails = attachment.get( 'cropDetails' );
+		const imgSelectOptions = this.imgSelect.getOptions();
+
+		// Account for imprecise cropping at minWidth/minHeight by snapping to minimum dimension if within 10 pixels.
+		if ( Math.abs( cropDetails.width - imgSelectOptions.minWidth ) < 10 ) {
+			cropDetails.width = imgSelectOptions.minWidth;
+		}
+		if ( Math.abs( cropDetails.height - imgSelectOptions.minHeight ) < 10 ) {
+			cropDetails.height = imgSelectOptions.minHeight;
+		}
+
+		// Force the resulting image to have the expected dimensions (scale down).
+		cropDetails.dst_width = imgSelectOptions.minWidth;
+		cropDetails.dst_height = imgSelectOptions.minHeight;
```

Comment:
Actually, we might as well let these be copies of the width/height. It is fine to have the featured image be _larger_ than the minimum.